src/FakeItEasy/Configuration/IThenConfiguration.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace FakeItEasy.Configuration
+{
+    using System.Diagnostics.CodeAnalysis;
+
+    /// <summary>
+    /// Configures the next behavior for this call.
+    /// </summary>
+    /// <typeparam name="TInterface">The type of configuration interface to return.</typeparam>
+    public interface IThenConfiguration<out TInterface>

```

Comment:
I hate to add a late-breaking comment, but shouldn't this hide object members?
So could many other interfaces in the review, and probably others. I only highlight this one because it's completely new.
On the other hand, we could just survey the exposed types and hide their object members later.